qualtran/bloqs/data_loading/qrom.py

```diff
@@ -136,20 +137,21 @@ def _load_nth_data(
             assert all(isinstance(x, (int, numbers.Integral)) for x in target_shape)
             for idx in np.ndindex(cast(Tuple[int, ...], target_shape)):
                 data_to_load = int(d[selection_idx + idx])
-                for q, bit in zip(target[idx], f'{data_to_load:0{target_bitsize}b}'):
-                    if int(bit):
-                        yield gate(q)
+                yield XorK(QUInt(target_bitsize), data_to_load).on(*target[idx]).controlled_by(
+                    *ctrl_qubits
+                )
+                # for q, bit in zip(target[idx], f'{data_to_load:0{target_bitsize}b}'):
```

Comment:
remove commented out code? OR say equivalent to if it's helpful for readability

Reply:
Had to remove, forgot. Had commented out assuming there would be test failures that I may need to go and debug; fortunately; there were no test failures :)